### PR TITLE
Docker: don't run as root user

### LINKED ISSUE #1779: Docker user should not run as root
See https://www.howtogeek.com/devops/why-processes-in-docker-containers-shouldnt-run-as-root/

Best to use a UID >10.000 range: https://elastisys.com/howto-stop-running-containers-as-root-in-kubernetes/ (also see https://medium.com/@gggauravgandhi/uid-user-identifier-and-gid-group-identifier-in-linux-121ea68bf510)

### PR DESCRIPTION
**Pull request by gerardsn**
closes #1779 

In this PR:
- default user in the docker container changed to `18081`
- moved and updated documentation for running docker
- set default container working dir to `/nuts`
- set defaults for various config paths to `./config/...` which is relative to the working directory
- fixed permission issues in e2e-tests. In most tests this means the `datadir` volume mount is removed if not needed

In a followup PR I will fix all default paths in the e2e-tests to cleanup a bit

TODO: 
- [ ] cleanup https://nuts-node.readthedocs.io/en/feature-dont-run-container-as-root/
- [ ] set same workdir in Dockerfile of dev image

### CODE REVIEW

**Comment by gerardsn** [comment]
yes, and that's why it is v6 and we need to add instructions on how to upgrade. I can't get it to fail on my laptop though

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: docs/pages/deployment/docker.rst [hunk 1]
```diff
@@ -0,0 +1,129 @@
+.. _running-docker:
+
+Running on Docker
+#################
+
+If you already use Docker, the easiest way to get your Nuts Node up and running for development or production is
+using Docker. This guide helps you to configure the Nuts node in Docker.
+To use the most recent release use ``nutsfoundation/nuts-node:latest``. For production environments it's advised to use a specific version.
+
```
Comment (reinkrul) [inline docs/pages/deployment/docker.rst:9]: can we start with the most important part when people start with Nuts, which is i.m.o. the Docker run and docker compose examples?

Then probably the dev image, and user/mount stuff

File: docs/pages/deployment/docker.rst [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+User
+****
+
+The default user in the container is ``18081`` that is only part of group ``18081``.
```
Comment (reinkrul) [inline docs/pages/deployment/docker.rst:13]: Should we note this is not relevant when not mounting the data dir (e.g. only `did:web` DID and storage in Postgres/Mysql)?

File: docs/pages/deployment/docker.rst [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+.. note::
+
+    - Nodes running the :ref:`recommended deployment <nuts-node-recommended-deployment>` (external storage configured for ``crypto.storage`` and ``storage.sql.connection``) that do not use did:nuts / gRPC network don't need to mount a ``data`` dir.
+
+    - *"User 18081 already exists on my host."* See `docker security <https://docs.docker.com/engine/security/userns-remap/>`_ (or relevant container orchestration platform) documentation how to restrict privileges to a user namespace / create a user mapping between host and container.
+
+Examples
+********
+The examples assume the current working directory on the host contains te following structure:
```
Comment (reinkrul) [inline docs/pages/deployment/docker.rst:50]:
```suggestion
The examples assume the current working directory on the host contains the following structure:
```

File: docs/pages/release_notes.rst [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
 
 **Release date:** TBD
-**Full Changelog**: https://github.com/nuts-foundation/nuts-node/compare/v5.0.0...master
+**Full Changelog**: https://github.com/nuts-foundation/nuts-node/compare/v5.0.0...v6.0.0
+
+Breaking
+********
+
+- Container user has changed from root to 18081.
+  When migrating from v5, change the owner of the data directory on the host to that of the container's user. (``chown -R 18081:18081 /path/to/host/data-dir``)
```
Comment (reinkrul) [inline docs/pages/release_notes.rst:16]: link/refer to docs?

File: docs/pages/release_notes.rst [hunk 5]
```diff
@@ -40,21 +45,25 @@ HTTP interface
...
 To simplify HTTP configuration and proxying and make the default more secure, HTTP endpoints now map to 2 HTTP interfaces:
 
-- port `8081` for all internal-facing endpoints (`/internal`, `/status`, `/metrics`, `/health`)
-- port `8080` for all public-facing endpoints (all others)
+- port ``8081`` for all internal-facing endpoints (``/internal``, ``/status``, ``/metrics``, ``/health``)
+- port ``8080`` for all public-facing endpoints (all others)
 
 The new HTTP configuration reflects this,
 
+<<<<<<< HEAD
```
Comment (reinkrul) [inline docs/pages/release_notes.rst:62]: conflicts

**Review by woutslakhorst — CHANGES_REQUESTED** [review]

File: docs/pages/deployment/docker.rst [hunk 6]
```diff
@@ -0,0 +1,129 @@
+.. _running-docker:
+
+Running on Docker
+#################
+
+If you already use Docker, the easiest way to get your Nuts Node up and running for development or production is
+using Docker. This guide helps you to configure the Nuts node in Docker.
+To use the most recent release use ``nutsfoundation/nuts-node:latest``. For production environments it's advised to use a specific version.
+
+User
```
[the remark this answers is not in the document]
Comment (gerardsn) [reply docs/pages/deployment/docker.rst:10]: The only documentation I can find has a different angle on managing users, so I added a note on L45:
>     - *"User 18081 already exists on my host."* See `docker security <https://docs.docker.com/engine/security/userns-remap/>`_ (or relevant container orchestration platform) documentation how to restrict privileges to a user namespace / create a user mapping between host and container.

File: docs/pages/release_notes.rst [hunk 7]
```diff
@@ -1,14 +1,19 @@
...
 #############
 
 *******************
-Peanut (v6 alpha 1)
+Peanut (v6.0.0)
 *******************
 
 **Release date:** TBD
-**Full Changelog**: https://github.com/nuts-foundation/nuts-node/compare/v5.0.0...master
+**Full Changelog**: https://github.com/nuts-foundation/nuts-node/compare/v5.0.0...v6.0.0
```
[the remark this answers is not in the document]
Comment (gerardsn) [reply docs/pages/release_notes.rst:10]: it still takes you to a page where you can manually set a different target. I'd rather set this correct now so we don't forget to change it on release.

File: docs/pages/deployment/docker.rst (hunk 1 above)
Comment (gerardsn) [reply docs/pages/deployment/docker.rst:8]: reordered some stuff, does this work?

**Comment by gerardsn** [comment]
Should the dev-container have the same user? consistency makes for better development, but fixing permission issues every time will be annoying.

**Comment by gerardsn** [comment]
Fixed issue introduced by this PR: `address` and `directory` are mutual exclusive config values for `policy`. This means you need to remove the default `directory` to use `address`. This is fixed in https://github.com/nuts-foundation/nuts-node/pull/2917/commits/60910951537c920057f17575c66809cab2b39210

File: docs/pages/deployment/docker.rst [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+Docker ``run``
+^^^^^^^^^^^^^^
+
+If you want to run without Docker Compose you can use the following command from the working directory:
+
+.. code-block:: shell
+
+  docker run --name nuts -p 8080:8080 -p 8081:8081 \
+    --mount type=bind,source="$(pwd)",target=/nuts \
+    nutsfoundation/nuts-node:latest
```
Comment (woutslakhorst) [inline docs/pages/deployment/docker.rst:65]: I would expect this one of the first things to see on this page.

**Review by woutslakhorst — APPROVED** [review]

File: docs/pages/deployment/docker.rst (hunk 8 above)
Comment (reinkrul) [reply docs/pages/deployment/docker.rst:65]: This

Comment (gerardsn) [reply docs/pages/deployment/docker.rst:65]: do you want to swap `Docker run` and `Docker Compose`, or is it about moving the context (folder structure / nuts.yaml) further down

File: docs/pages/deployment/docker.rst [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+*************
+
+The default working directory within the container is ``/nuts`` that provides defaults for the various configurable data and config paths used:
+
+.. code-block:: none
+
+    nuts
+    ├── config
+    │   ├── discovery/...  <discovery.definitions.directory>
+    │   ├── jsonld/...     <>
```
Comment (reinkrul) [inline docs/pages/deployment/docker.rst:94]: `<>`?

File: docs/pages/deployment/docker.rst [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+****
+
+The default user in the container is ``18081`` that is only part of group ``18081``.
+This is a regular user without root privileges to provide an additional level of security.
+If ``datadir`` config value points to a mounted directory, see the section below how to manage privileges needed by the nuts-node.
+
+Volume mounts
+*************
+
+The default working directory within the container is ``/nuts`` that provides defaults for the various configurable data and config paths used:
```
Comment (reinkrul) [inline docs/pages/deployment/docker.rst:87]: Not sure whether this section (mapping config to directories) adds anything for people who want to use it? The (default) paths should be in the configuration reference, and the options and directory names self-explanatory?

Less = more?

File: docs/pages/release_notes.rst [hunk 11]
```diff
@@ -1,14 +1,19 @@
-
 #############
 Release notes
 #############
 
 *******************
-Peanut (v6 alpha 1)
+Peanut (v6.0.0)
```
Comment (reinkrul) [inline docs/pages/release_notes.rst:6]: we should make more clear this version isn't released yet, so maybe omit the version/reprhase it?

File: docs/pages/deployment/docker.rst [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+.. code-block:: none
+
+    .
+    ├── config
+    │   └── nuts.yaml
+    └── data
+
+The ``data`` directory must be owned by container user ``18081``. (``chown -R 18081:18081 ./data``)
+
+These examples use the following ``nuts.yaml`` configuration file:
```
Comment (reinkrul) [inline docs/pages/deployment/docker.rst:22]: Maybe we should simplify this by not using the YAML file, but environment variables?

**Review by reinkrul — APPROVED** [review]
Functional changes approved, but I find the documentation too complicated (we're in the risk of getting too much of it, again)

File: docs/pages/release_notes.rst (hunk 11 above)
Comment (gerardsn) [reply docs/pages/release_notes.rst:6]: L9 
> **Release date:** TBD

File: docs/pages/deployment/docker.rst (hunk 9 above)
Comment (gerardsn) [reply docs/pages/deployment/docker.rst:94]: removed

**Pull request merged**